rusk/src/lib/encoding.rs

```diff
@@ -176,9 +178,46 @@ impl TryFrom<&mut rusk_proto::Transaction> for Transaction {
     }
 }
 
-impl From<SpentTransaction> for rusk_proto::Transaction {
-    fn from(spent_tx: SpentTransaction) -> Self {
-        let tx = spent_tx.0;
+impl From<Error> for rusk_proto::executed_transaction::Error {
+    fn from(err: Error) -> Self {
+        use rusk_proto::executed_transaction::error::Code;
+
+        let (code, contract_id, data) = match err {
+            Error::Vm(e) => match e {
+                VMError::UnknownContract(id) => {
+                    (Code::UnknownContract, id, format!("{}", e))
+                }
+                VMError::ContractPanic(id, data) => {
+                    (Code::ContractPanic, id, data)
+                }
+                VMError::OutOfGas => (
+                    Code::OutOfGas,
+                    rusk_abi::transfer_contract(),
```

Comment:
What happen if we run out of gas during a stake? It will result as transfer_contract's error?

Reply:
Yes, because the `OutOfGas` is not related to the contract, is related to the transaction, and all the transaction are initiated by the transfer contract's `execute`. So basically the block explorer won't don't display that "this contract" is went out of gas, but that the transaction didn't have enough gas to go through.

Different story is for `UnknownContract` and `ContractPanic`, since if you have contract `A` calling `B` calling `C` you want to know which of those is panicking or it's unknown.